diff --git a/parse.c b/parse.c
index c536ec2..c698661 100644
--- a/parse.c
+++ b/parse.c
@@ -1140,6 +1140,11 @@ static void end_iterator(struct statemen
 	end_symbol_scope();
 }
 
+static struct symbol_list *fn_stack;
+#define push_fn(sym)	add_ptr_list(&fn_stack,sym)
+#define pop_fn()	({delete_ptr_list_last((struct ptr_list **)&fn_stack); \
+			first_ptr_list((struct ptr_list *)fn_stack);})
+
 static struct statement *start_function(struct symbol *sym)
 {
 	struct symbol *ret;
@@ -1156,13 +1161,14 @@ static struct statement *start_function(
 
 	// Currently parsed symbol for __func__/__FUNCTION__/__PRETTY_FUNCTION__
 	current_fn = sym;
+	push_fn(sym);
 
 	return stmt;
 }
 
 static void end_function(struct symbol *sym)
 {
-	current_fn = NULL;
+	current_fn = pop_fn();
 	end_function_scope();
 }
 
